src/entry_queue.rs

```diff
@@ -0,0 +1,149 @@
+use crate::poh_service::NUM_TICKS_PER_SECOND;
+use hashbrown::HashMap;
+use solana_sdk::hash::Hash;
+use solana_sdk::timing::timestamp;
+
+/// The number of most recent `last_id` values that the bank will track the signatures
+/// of. Once the bank discards a `last_id`, it will reject any transactions that use
+/// that `last_id` in a transaction. Lowering this value reduces memory consumption,
+/// but requires clients to update its `last_id` more frequently. Raising the value
+/// lengthens the time a client must wait to be certain a missing transaction will
+/// not be processed by the network.
+pub const MAX_ENTRY_IDS: usize = NUM_TICKS_PER_SECOND * 120;
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct Entry {
```

Comment:
this looks like locktower coming in?  can we use a different name for these?

Reply:
How about LastId queue?